nixos/modules/system/boot/systemd/sysupdate.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.systemd.sysupdate;
+
+  writeDefinition = name: transferDefinition: pkgs.writeText
+    "${name}.conf"
+    (lib.generators.toINI { } transferDefinition);
+
+  listOfDefinitions = lib.mapAttrsToList
+    writeDefinition
+    (lib.filterAttrs (k: _: !(lib.hasPrefix "_" k)) cfg.transfers);
+
+  definitionsDirectory = pkgs.runCommand "systemd-sysupdate-definitions" { } ''
+    mkdir -p $out
+    ${(lib.concatStringsSep "\n"
+      (map (pkg: "cp ${pkg} $out/${pkg.name}") listOfDefinitions)
+    )}
+  '';
+in
+{
+  options.systemd.sysupdate = {
+
+    enable = lib.mkEnableOption (lib.mdDoc "systemd-sysupdate") // {
+      description = lib.mdDoc ''
+        Atomically update the host OS, container images, portable service
+        images or other sources.
+      '';
+    };
+
+    transfers = lib.mkOption {
+      type = with lib.types; attrsOf (attrsOf (attrsOf (oneOf [ str int bool ])));
```

Comment:
Do we want to use `submodule` here to make this type a bit more precise?

Reply:
I have a weak preference to keep it as a RFC-42 structural setting. Then we don't have to change the module if systemd ever introduces another section. If you have a strong preference, however, I'll gladly add the sections via a submodule.

Reply:
The type right now is not that helpful on its own, it could be almost anything

Reply:
The more I think about it the stronger my preference for this structural setting style gets. Any user that wants to use this module has to look at the upstream docs anyways to figure out what options to set. So there is little value in adding more type safety on the Nix side to it but a lot of maintenance burden. This is a drawback that RFC-42 explicitly tries to address.